### ISSUE TITLE
IndexRedirect not passing query

### ISSUE THREAD

**Issue by gabrielenosso** [rackt/redux-simple-router#215]
In an example like the one below, the IndexRedirect is ignoring the "query" prop:

```js
<Router history={history}>
   <Route path="/" component={App}>
     <Route path="events" component={EventsPage} />
     <Route path="masterdata" component={MasterDataPage} />
     <IndexRedirect to={'/events'} query={{ player: 23 }} />
   </Route>
</Router>
```

The action UPDATE_LOCATION is created, but the location.query is an empty object.

It works fine if I use a common <Link> component though.

**Comment by timdorr** [comment]
Do you have an example repo I can look at? I can test this, but it's still a lot of setup work...

**Comment by Aweary** [comment]
What version of `history` are you using?

**Comment by gabrielenosso** [comment]
I'm using the latest one ("history": "^2.0.0-rc2").


I created the history just following the doc:

// Create history for react router and connect it to the redux store
const history = useRouterHistory(createHistory)({
  queryKey: false,
  parseQueryString: urlQueryCsv.parse,
  stringifyQuery: urlQueryCsv.stringify
});

// Sync dispatched route actions to the history
const reduxRouterMiddleware = syncHistory(history);

// Add all middlewares
const createStoreWithMiddleware = compose(
  applyMiddleware(
    reduxRouterMiddleware,
    thunk
  ),
  window.devToolsExtension ? window.devToolsExtension() : f => f
)(createStore);

// Create redux store
const store = createStoreWithMiddleware(reducer);

// Required for replaying actions from devtools to work
reduxRouterMiddleware.listenForReplays(store);

// Use redux store with react-redux Provider
render(
  <Provider store={store}>
    <Routes history={history} />
  </Provider>,
  document.getElementById('root')
);

**Comment by timdorr** [comment]
Ah, this is a react-router bug: rackt/react-router#2917 It will be fixed one we get another history release (2.0.0-rc3, most likely) and get a react-router release incorporating that release. Unfortunately, we'll just have to wait on that, but hopefully it can be done soon.
